### PR TITLE
Add types for "desktop-service"

### PR DESCRIPTION
**Pull request by tariibaba**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/desktop-service/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        "strictFunctionTypes": true,
+        "strictNullChecks": true,
+        "baseUrl": "../",
+        "typeRoots": [
+            "../"
+        ],
+        "types": ["node"],
+        "noEmit": true,
+        "forceConsistentCasingInFileNames": true,
+        "esModuleInterop": true,
```
Comment (peterblazejewicz) [inline types/desktop-service/tsconfig.json:18]:
```suggestion
        "forceConsistentCasingInFileNames": true
```

Comment (peterblazejewicz) [reply types/desktop-service/tsconfig.json:18]: this is not adviced on DT for interop, also there was a syntax error (it's not JSONC)

File: types/desktop-service/desktop-service-tests.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import Service from './';
```
Comment (peterblazejewicz) [inline types/desktop-service/desktop-service-tests.ts:1]:
```suggestion
import Service = require('desktop-service');
```

File: types/desktop-service/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,107 @@
+// Type definitions for desktop-service 1.0
+// Project: https://github.com/tariibaba/desktop-service#readme
+// Definitions by: Ayibatari Ibaba <https://github.com/me>
```
Comment (peterblazejewicz) [inline types/desktop-service/index.d.ts:3]:
```suggestion
// Definitions by: Ayibatari Ibaba <https://github.com/tariibaba>
```

File: types/desktop-service/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+            "../"
+        ],
+        "types": ["node"],
+        "noEmit": true,
+        "forceConsistentCasingInFileNames": true
+    },
+    "files": [
+        "index.d.ts",
+        "desktop-service-tests.ts"
+    ],
```
Comment (peterblazejewicz) [inline types/desktop-service/tsconfig.json:22]:
```suggestion
    ]
```

File: types/desktop-service/tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+        ],
+        "noImplicitAny": true,
+        "noImplicitThis": true,
+        "strictFunctionTypes": true,
+        "strictNullChecks": true,
+        "baseUrl": "../",
+        "typeRoots": [
+            "../"
+        ],
+        "types": ["node"],
```
Comment (peterblazejewicz) [inline types/desktop-service/tsconfig.json:15]:
```suggestion
        "types": [],
```

File: types/desktop-service/index.d.ts [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+    /**
+     * Setting this to `true` will force the process to exit if it encounters
+     * an error that stops the node.js script from running.
+     * This does not mean the process will stop if the script throws an error.
+     * It will only abort if the script throws an error causing the process to exit
+     * (i.e. `process.exit(1)`).
+     */
+    abortOnError?: boolean;
+    /** The initial number of seconds to wait before attempting a restart
+     * (after the script stops). */
```
Comment (peterblazejewicz) [inline types/desktop-service/index.d.ts:35]:
```suggestion
     * (after the script stops).
     */
```

File: types/desktop-service/index.d.ts [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+    maxRetries?: number;
+    /**
+     * Setting this to `true` will force the process to exit if it encounters
+     * an error that stops the node.js script from running.
+     * This does not mean the process will stop if the script throws an error.
+     * It will only abort if the script throws an error causing the process to exit
+     * (i.e. `process.exit(1)`).
+     */
+    abortOnError?: boolean;
+    /** The initial number of seconds to wait before attempting a restart
```
Comment (peterblazejewicz) [inline types/desktop-service/index.d.ts:34]:
```suggestion
    /**
     *The initial number of seconds to wait before attempting a restart
```

File: types/desktop-service/index.d.ts [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+    start(): void;
+    /** Stop the service if it is currently running. */
+    stop(): void;
+    /** Restart the service. */
+    restart(): void;
+    /** The absolute path of the script to launch as a service. */
+    script: string;
+    /** The name of the service. */
+    name: string;
+    /** The initial number of seconds to wait before attempting a restart
```
Comment (peterblazejewicz) [inline types/desktop-service/index.d.ts:82]:
```suggestion
    /**
     * The initial number of seconds to wait before attempting a restart
```

File: types/desktop-service/index.d.ts [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+     * (after the script stops). */
```
Comment (peterblazejewicz) [inline types/desktop-service/index.d.ts:83]:
```suggestion
     * (after the script stops).
     */
```

File: types/desktop-service/index.d.ts [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+     * It will only abort if the script throws an error causing the process to exit
+     * (i.e. `process.exit(1)`).
+     */
+    abortOnError: boolean;
+    on(event: 'install', listener: () => void): this;
+    on(event: 'alreadyinstalled', listener: () => void): this;
+    on(event: 'invalidinstallation', listener: () => void): this;
+    on(event: 'uninstall', listener: () => void): this;
+    on(event: 'start', listener: () => void): this;
+    on(event: 'stop', listener: () => void): this;
```
Comment (peterblazejewicz) [inline types/desktop-service/index.d.ts:98]:
```suggestion
    on(event: 'install' | 'alreadyinstalled' | 'invalidinstallation' | 'uninstall' | 'start' | 'stop', listener: () => void): this;
```

File: types/desktop-service/index.d.ts (hunk 9 above)
Comment (peterblazejewicz) [inline types/desktop-service/index.d.ts]:
```suggestion
     * (after the script stops).
     */
```

File: types/desktop-service/index.d.ts [hunk 11]
```diff
@@ -0,0 +1,106 @@
...
+     * It will only abort if the script throws an error causing the process to exit
+     * (i.e. `process.exit(1)`).
+     */
+    abortOnError: boolean;
+    on(event: 'install' | 'alreadyinstalled' | 'invalidinstallation' | 'uninstall' | 'start' | 'stop', listener: () => void): this;
+    on(event: 'error', listener: (err: any) => void): this;
+    on(event: string | symbol, listener: (...args: any[]) => void): this;
+}
+
+export = Service
```
Comment (peterblazejewicz) [inline types/desktop-service/index.d.ts:101]:
```suggestion
export = Service;
```

File: types/desktop-service/index.d.ts [hunk 12]
```diff
@@ -0,0 +1,106 @@
...
+
+interface EnvironmentVariables {
+    name: string;
+    value: any;
+}
```
Comment (peterblazejewicz) [inline types/desktop-service/index.d.ts:106]:
```suggestion
}

```

**Review by peterblazejewicz — APPROVED** [review]
LGMT!
@tariibaba I've made some minor tweaks with code (module setup, formatting, etc), no implementation changes.

**Comment by tariibaba** [comment]
Ready to merge

**Pull request merged**